src/state.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <stdint.h>
+
+#include "./hash_table.h"
+
+/* Struct representing a connected client. */
+typedef struct connected_client {
+  int file_descriptor;
+  uint16_t port;
+} connected_client;
+
```

Comment:
I don't think we can easily do this. When a client connects to us, we don't know what port they are serving on 
It would be easier to separate the two and have two structures (one for known ports clients are listening on and one for file_descriptors that we are connected to)